src/main/java/sparksoniq/jsoniq/item/Item.java

```diff
@@ -154,6 +154,17 @@ else if (v1 instanceof NullItem) {
             else{
                 result = 1;
             }
+        } else if (v1 == null || v2 == null) {
```

Comment:
While this is correct, I would rather have this code to compare Java nulls in compareWithFlowrKey() rather than here.

Indeed, the fact that null represents the empty sequence is an internal implementation detail of FlworKeys that we should "protect" the Item class from, for cleanliness. The Item class only deals with single items. This will avoid surprises in the future.